pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationDataBasic.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.client.impl.auth;
+
+import org.apache.pulsar.client.api.AuthenticationDataProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import java.util.Base64;
+
+public class AuthenticationDataBasic implements AuthenticationDataProvider {
+    private final static String HTTP_HEADER_NAME = "Authorization";
+    private String authToken;
+
+    public AuthenticationDataBasic(String userId, String password) {
+        authToken = "Basic " + Base64.getEncoder().encodeToString((userId + ":" + password).getBytes());
```

Comment:
If I remember correctly, authentication method name is stored in some field of  a command. "Basic " is needed for the HTTP header but it's redundant for the pulsar's command.

Also, base64 encoding is needed because of a limitation of HTTP headers. I don't think pulsar's command has the same limitation.

They both increases data length unnecessarily。

Reply:
I made command data not contain encoded params.